tests/bluetooth/bsim_bt/bsim_test_rpa_central/src/tester.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bs_bt_utils.h"
+#include "zephyr/bluetooth/addr.h"
+#include "zephyr/bluetooth/bluetooth.h"
+#include "zephyr/bluetooth/conn.h"
+#include "zephyr/toolchain/gcc.h"
+#include <host/hci_core.h>
+
+#include <stdint.h>
+#include <string.h>
+
+DEFINE_FLAG(flag_new_address);
+
+static struct bt_le_adv_param params;
+static struct bt_le_ext_adv_start_param start_params;
+static struct bt_le_ext_adv *adv;
+
+void scanned_cb(struct bt_le_ext_adv *adv, struct bt_le_ext_adv_scanned_info *info)
+{
+	static bool init;
+	static bt_addr_le_t old_addr, new_addr;
+	static uint64_t old_time, new_time, diff;
+
+	if (bst_result == Passed) {
+		return;
+	}
+
+	if (!init) {
+		old_addr = *info->addr;
+		old_time = k_uptime_get();
+		init = true;
+	}
+
+	new_addr = *info->addr;
+
+	/* Check if the scan request comes from a new address */
+	if (bt_addr_le_cmp(&old_addr, &new_addr)) {
+		printk("Scanned request from new ");
+		print_address(info->addr);
+
+		/* Ensure the RPA rotation occurs within +-10% of CONFIG_BT_RPA_TIMEOUT */
+		new_time = k_uptime_get();
+		if ((new_time - old_time) > (CONFIG_BT_RPA_TIMEOUT * 1000)) {
+			diff = (new_time - old_time) - (CONFIG_BT_RPA_TIMEOUT * 1000);
+		} else {
+			diff = (CONFIG_BT_RPA_TIMEOUT * 1000) - (new_time - old_time);
+		}
```

Comment:
Suggest to use some variables here:

```suggestion
		new_time = k_uptime_get();
		time_diff_ms = new_time - old_time;
		rpa_timeout_ms = CONFIG_BT_RPA_TIMEOUT * MSEC_PER_SEC;
		
		if (time_diff_ms > rpa_timeout_ms) {
			diff = time_diff_ms - rpa_timeout_ms;
		} else {
			diff = rpa_timeout_ms - time_diff_ms;
		}
```